build_projects/shared-build-targets-utils/VersionRepoUpdater.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.DotNet.Cli.Build
+{
+    public class VersionRepoUpdater
+    {
+        private static Regex s_nugetFileRegex = new Regex("^(.*?)\\.(([0-9]+\\.)?[0-9]+\\.[0-9]+(-([A-z0-9-]+))?)\\.nupkg$");
+
+        private HttpClient _client = new HttpClient();
```

Comment:
Initializing this as part of `UpdatePublishedVersions` would make it a bit easier to dispose this.